2-5 add a new space with name/desc/price

MisaOgura opened the pull request:
Closes #2 
Closes #3 
Closes #4 
Closes #5

MisaOgura committed:
Setup DataMapper, create test database, pass first test for adding a space to the database

MisaOgura committed:
Set up database cleaner, add name property to spaces table

MisaOgura committed:
Add description property to spaces table

MisaOgura committed:
Add price property to spaces table

gvonkoss commented on app/models/space.rb:
```diff
@@ -8,6 +7,7 @@ class Space
   property :id, Serial
   property :name, String
   property :description, String
```
Perhaps this should be Text, considering character limitations?

gvonkoss commented on app/spec/space_spec.rb:
```diff
@@ -0,0 +1,26 @@
+                 price: 75)
+    expect(Space.first.price).to eq(75)
+  end
+end
```
I think we could use doubles for name, space and description here!

MisaOgura committed:
User can see name, descriotion and price forms plus submit button on page /spcae/new

MisaOgura committed:
Spec test using doubles, now drier

MisaOgura committed:
Creating interface.js for feature testing the list page

MisaOgura committed:
Modify the file structure of public directory

gvonkoss commented on app/spec/space_spec.rb:
```diff
@@ -0,0 +1,26 @@
+    Space.create(name: 'space1',
+                 description: 'One bedroom flat',
+                 price: 75)
+  end
```
I would still use doubles!
e.g.

```
let(:name) { 'space1' }
let(:space) { Space.first }
...
expect (space.name).to eq name
```

something along those lines!

gvonkoss commented on app/spec/spec_helper.rb:
```diff
@@ -1,10 +1,11 @@
 require 'capybara'
 require 'capybara/rspec'
 require 'rspec'
+require 'database_cleaner'
```
Just to avoid merging conflicts, please include `require data_mapper` just before `require database_cleaner` thaaaanks

MisaOgura committed:
Trying to make jQuery work on capybara testing environment. Development environment working

MisaOgura committed:
Reorganise file structure accorging to the lates master branch

MisaOgura committed:
Ajax get request for details of one space implemented

MisaOgura committed:
Separate driver of use between unit test (:selenium) and feature test (:js)
Modify spec_helper config to reflect database cleaning to feature test which now uses different driver.

MisaOgura commented:
A step away from making AJAX request work in the feature test, trying a helper method listed here: https://robots.thoughtbot.com/automatically-wait-for-ajax-with-capybara

MisaOgura committed:
Implement Capybara helper method to wait for AJAX request completion. File required in spec_helper

MisaOgura committed:
Moved requirement for helper and model to spec_helper file. AJAX in feature test still not rendering the grabbed information

MisaOgura committed:
Add wait_for_ajax function in feature test

MisaOgura committed:
Install poltergeist driver for capybara feature test. Not working 100% but will leave it as it is

gvonkoss committed:
fixed merge conflicts, refactored test

MisaOgura commented:
:shipit:
